cachalot/utils.py

```diff
@@ -178,18 +179,32 @@ def __update_annotated_subquery(_annotation: Subquery):
             else:
                 tables.update(_get_tables(db_alias, _annotation.query))
 
+        def flatten(expression: BaseExpression):
+            """
+            Recursively yield this expression and all subexpressions, in
+            depth-first order.
+
+            Taken from Django 3.2 as the previous Django versions don’t check
+            for existence of flatten.
+            """
+            yield expression
+            for expr in element.get_source_expressions():
+                if expr:
+                    if hasattr(expr, 'flatten'):
+                        yield from flatten(expr)
+                    else:
+                        yield expr
+
         # Gets tables in subquery annotations.
         for annotation in query.annotations.values():
-            if isinstance(annotation, Case):
-                for case in annotation.cases:
-                    for subquery in _find_subqueries_in_where(case.condition.children):
-                        tables.update(_get_tables(db_alias, subquery))
-                if isinstance(annotation.default, Subquery):
-                    __update_annotated_subquery(annotation.default)
-            elif isinstance(annotation, Subquery):
-                __update_annotated_subquery(annotation)
-            elif type(annotation) in UNCACHABLE_FUNCS:
+            if type(annotation) in UNCACHABLE_FUNCS:
                 raise UncachableQuery
+            for element in flatten(annotation):
+                if isinstance(element, Subquery):
+                    __update_annotated_subquery(element)
+                elif isinstance(element, RawSQL):
+                    sql = repr(element).lower()
```

Comment:
Why not use `as_sql`? I thought the repr includes the class name: https://github.com/django/django/blob/bf5abf1bdcedb15e949db419c61eeec7c88414ea/django/db/models/expressions.py#L812

Reply:
@Andrew-Chen-Wang: yes. Instead of calling `as_sql` with fake arguments (as they are not used in the method) we could also directly access the sql attribute of `RawSQL`. https://github.com/django/django/blob/f42ccdd835e5b3f0914b5e6f87621c648136ea36/django/db/models/expressions.py#L815

What do you reckon?